pkg/server/home_workspaces.go

```diff
@@ -351,6 +378,21 @@ func (h *homeWorkspaceHandler) ServeHTTP(rw http.ResponseWriter, req *http.Reque
 				http.Error(rw, "Creating the home workspace", http.StatusTooManyRequests)
 				return
 			}
+
+			if requestInfo.IsResourceRequest && requestInfo.APIGroup == tenancyv1alpha1.SchemeGroupVersion.Group && requestInfo.Resource == "clusterworkspaces" {
```

Comment:
This kind of breaks my brain - can we close over this in `searchForWorkspaceAndRBACInLocalInformers`? Otherwise the flow here is pretty hard to follow (and we need to discern specifics about what might or might be true based on what some other helper returned)

Reply:
I don't have the `requestInfo` in `searchForWorkspaceAndRBACInLocalInformers` and it would be a huge amount of work to fix all the test cases that would be broken by adding a param to that function. That's why I did it here.